components/Common/Pagination/useGetPageElements.tsx

```diff
@@ -0,0 +1,146 @@
+'use client';
+import Link from 'next/link';
+import { useIntl } from 'react-intl';
+
+import styles from './index.module.css';
+
+import type { PaginationProps } from '.';
+
+type ParsedPage = PaginationProps['pages'][number] & {
+  pageNumber: number;
+};
+
+function parsePages(pages: PaginationProps['pages']): ParsedPage[] {
+  return pages.map(({ url }, index) => ({
+    url,
+    pageNumber: index + 1,
+  }));
+}
+
+const ellipsis = (
+  <span aria-hidden="true" className={styles.ellipsis}>
+    ...
+  </span>
+);
+
+function createPageElements({
```

Comment:
This... should probably be a Component on its own.

Also, this Hook is sorta overcomplicated

Reply:
I agree with you in that it could be a component on its own. A sub-component called `PaginationListItem`, for example, with the following markup:

```tsx
  <li
        key={pageNumber}
        aria-setsize={parsedPages.length}
        aria-posinset={pageNumber}
      >
        <Link
          prefetch={false}
          href={url}
          aria-label={intl.formatMessage(
            {
              id: 'components.common.pagination.pageLabel',
            },
            {
              pageNumber,
            }
          )}
          className={styles.listItem}
          {...(pageNumber === currentPage && { 'aria-current': 'page' })}
        >
          <span>{pageNumber}</span>
        </Link>
      </li>

```

However, as per the complexity of the hook, the thing is that it's inherently a complex logic, although I have tried my best through the tests and the code to make it as explicative as possible.

I will replace the magic numbers 3 and 2 and add more comments, overall. I personally avoid large comment blocks, but I guess that in this case they would be helpful.

Reply:
Right, also I believe this should indeed be a sub-component. Can you do that?